test/unit/configuration/configuration_finding_test.js

```diff
@@ -0,0 +1,232 @@
+      .mockFile(configInTempDirectory, 'random content');
+    mock('fs/promises', {
+      access: async function (path) {
+        return fsMock.access(path);
+      },
+      readFile: async function (path){
+        return fsMock.readFile(path);
+      }
+    });
+    mock('process', {
+      env: {
+        SF_CLIENT_CONFIG_FILE: configFromEnvVariable
+      }
+    });
+    const fsPromises = require('fs/promises');
+    const process = require('process');
+    const configUtil = new ConfigurationUtil(fsPromises, process);
+
+    // when
+    const configFound = await configUtil.getClientConfig(configFromConnectionString);
+
+    // then
+    assert.deepEqual(configFound, clientConfig);
+  });
+
+  it('should take config from environmental variable if no input present', async function () {
+    // given
+    const fsMock = new FsMock()
+      .mockFile(configFromEnvVariable, fileContent)
+      .mockFile(configInDriverDirectory, 'random content')
+      .mockFile(configInHomeDirectory, 'random content')
+      .mockFile(configInTempDirectory, 'random content');
+    mock('fs/promises', {
+      access: async function (path) {
+        return fsMock.access(path);
+      },
+      readFile: async function (path){
+        return fsMock.readFile(path);
+      }
+    });
+    mock('process', {
+      env: {
+        SF_CLIENT_CONFIG_FILE: configFromEnvVariable
+      }
+    });
+    const fsPromises = require('fs/promises');
+    const process = require('process');
+    const configUtil = new ConfigurationUtil(fsPromises, process);
+
+    // when
+    const configFound = await configUtil.getClientConfig(null);
+
+    // then
+    assert.deepEqual(configFound, clientConfig);
+  });
+
+  it('should take config from driver directory if no input nor environmental variable present', async function () {
+    // given
+    const fsMock = new FsMock()
+      .mockFile(configInDriverDirectory, fileContent)
+      .mockFile(configInHomeDirectory, 'random content')
+      .mockFile(configInTempDirectory, 'random content');
+    mock('fs/promises', {
+      access: async function (path) {
+        return fsMock.access(path);
+      },
+      readFile: async function (path){
+        return fsMock.readFile(path);
+      }
+    });
+    mock('process', {
+      env: {
+        SF_CLIENT_CONFIG_FILE: undefined
+      }
+    });
+    const fsPromises = require('fs/promises');
+    const process = require('process');
+    const configUtil = new ConfigurationUtil(fsPromises, process);
+
+    // when
+    const configFound = await configUtil.getClientConfig(null);
+
+    // then
+    assert.deepEqual(configFound, clientConfig);
+  });
+
+  it('should take config from home directory if no input nor environmental variable nor in driver directory present', async function () {
+    // given
+    const fsMock = new FsMock()
+      .mockFile(configInHomeDirectory, fileContent)
+      .mockFile(configInTempDirectory, 'random content');
+    mock('fs/promises', {
```

Comment:
this mock is repeating, we can extract it to function

Reply:
done